### ISSUE TITLE
operator_whitespace warning when overloading an operator

### ISSUE THREAD

**Issue by arnopoulos** [realm/SwiftLint#666]
The linter throws a warning when overloading an operator and not placing a space before the list of parameters.

Warning example:
-
```
func ==(left: SomeObject, right: SomeObject) { ... }
```

This should not throw a warning due to the fact that it is a function definition and the function definition rules allow you to either have a space or forego one.

**Comment by norio-nomura** [comment]
duplicate of #594?

**Comment by jpsim** [comment]
this is intended behavior following the GitHub Swift guidelines that SwiftLint is built for: https://github.com/github/swift-style-guide#use-whitespace-around-operator-definitions
